### ISSUE TITLE
npm@4 search crashes on Windows

### ISSUE THREAD

**Issue by fibric** [npm/npm#14362]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
I tried out version npm 4.0.0 because I'm a victim of #12619. So I initiated a search for `yarn`. At the end (but not sure if npm really found all results) npm crashed while trying to rename a cache file.
It also didn't create the log file which is mentioned at the end of the error output.

https://gist.github.com/fibric/dd1333056ac241cdb2d166cba4aff7a5

npm suggests re-running the same command but with Administrator privileges. No, I'll never do that. The cache file is located in my users folder, and never ever I run a command to mess around with Administrator privileges in my users folder on a Windows machine - I got bitten hard once and lesson I learned. (Unfortunately I have to use a corporate Windows machine.)

#### How can the CLI team reproduce the problem?

- `npm i -g npm@next`
- `npm search yarn`

### supporting information:

 - `npm -v` prints: `4.0.0`
 - `node -v` prints: `v6.9.1`
 - `npm config get registry` prints: `https://registry.npmjs.org/`
 - Windows, OS X/macOS, or Linux?: `Windows 7 64x Enterprise`
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by zkat** [comment]
@fibric The issue at hand is that for some reason, npm can't write to C:\\Users\\denny.trebbin\\AppData\\Roaming\\npm-cache\\registry.npmjs.org\\-\\all\\.cache.json, which would kinda be necessary for the search cache to be written.

There's not much I can do to help with this besides making sure that you have write permissions to that directory (and its parents), as the user you're running search with.

**Comment by zkat** [comment]
(For the record,  I did try this on my own Windows machine and was unable to reproduce it. There's also a fairly comprehensive test suite surrounding the new search implementation, so if there's a specific corner case you can isolate here, lmk. I'm not sure erroring out when a cache file cannot be written is a problem, though we have an upcoming error messaging improvement project)

**Comment by fibric** [comment]
The file is being written and has a size of 154MB
![image](https://cloud.githubusercontent.com/assets/623932/19800745/783c13c0-9cfc-11e6-85f8-b81f8540a1cb.png)

I deleted the cache.json file. Re-Issued another search `npm search babili` and got the same error.

In this picture, you'll see that it started writing the `.cache.json.437622700` file. Few seconds after search started the file size is 14MB
![image](https://cloud.githubusercontent.com/assets/623932/19800806/c4386850-9cfc-11e6-870d-8613070e750b.png)
In this picture, you'll see that npm renamed the file to `.cache.json` with a size of 201MB.
![image](https://cloud.githubusercontent.com/assets/623932/19800859/05f382ca-9cfd-11e6-8de7-31cc5b9a713f.png)
In this picture, you'll see that npm errors and prints a not so truthy error message.
![image](https://cloud.githubusercontent.com/assets/623932/19800848/fe8e2d78-9cfc-11e6-92d7-2d6cce61bd9e.png)

And again, no log file written to `C:\code\js\ci-node-test\npm-debug.log`
```
λ ls -Ahl
total 316K
-rw-r--r-- 1 denny.trebbin 1049089   98 Oct 20 12:19 .babelrc
-rw-r--r-- 1 denny.trebbin 1049089 1.2K Oct 20 12:19 .editorconfig
drwxr-xr-x 1 denny.trebbin 1049089    0 Oct 28 11:00 .git/
-rw-r--r-- 1 denny.trebbin 1049089  17K Oct 20 11:18 .gitignore
drwxr-xr-x 1 denny.trebbin 1049089    0 Oct 28 10:42 .next/
-rw-r--r-- 1 denny.trebbin 1049089   51 Oct 20 11:16 LICENSE.md
-rw-r--r-- 1 denny.trebbin 1049089   13 Oct 20 11:14 README.md
drwxr-xr-x 1 denny.trebbin 1049089    0 Oct 28 10:36 node_modules/
-rw-r--r-- 1 denny.trebbin 1049089  425 Oct 28 10:39 package.json
drwxr-xr-x 1 denny.trebbin 1049089    0 Oct 28 10:41 pages/
-rw-r--r-- 1 denny.trebbin 1049089 122K Oct 28 10:36 yarn.lock
```

**Comment by fibric** [comment]
- I uninstalled Node.js `choco uninstall nodejs.install`.
- Then deleted all the files and folders left untouched _(hell, I don't know why people complain about macOS uninstall procedures when at the same time, not even Windows uninstallers clean up correctly)._
- Then installed Node.js again `choco install nodejs.install`
- Then I started another search `npm search yarn`
Finally no error.

I don't know how it happened. Maybe Windows is just garbage; the enforced anti-virus may lock the file for too long (still, npm shouldn't fail) or it's just bad behavior when npm recognizes people looking for alternatives.
